### ISSUE TITLE
UselessParentheses false positive with array cast dereference

### ISSUE THREAD

**Issue by nishimura-d** [slevomat/coding-standard#1095]
```php
            $key = ((array)$table->getPrimaryKey())[0];
            // $key === 'id' if $table->getPrimaryKey() === 'id'
```

removed parentheses incorrectly:

```php
            $key = (array)$table->getPrimaryKey()[0];
            // $key === ['i'] if $table->getPrimaryKey() === 'id'
```

**Comment by kukulich** [comment]
Thank you for your report.

Fixed in https://github.com/slevomat/coding-standard/commit/a95e8e38172118d415247b8fb27b11030aae790c
